### ISSUE TITLE
Connection Pool for Client -> Out/Destination

### ISSUE THREAD

**Issue by Lol3rrr** [Lol3rrr/tunneler#1]
## Problem
Right now, for every new user connection the client creates a new TCP-Connection out to the Destination. This creates a pretty significant performance bottleneck that is otherwise hard to solve and especially impacts the latency.

## Solution
Create a custom connection pool, that will keep a cerain amount of connections open and ready for use as well as reuse connections after they are done no longer needed by a user. This could hopefully remove the latency of creating a new connection everytime.

**Comment by Lol3rrr** [comment]
The first implementation was just merged with [this merge request](https://github.com/Lol3rrr/tunneler/pull/2).

# Left to do
- [ ] Create a new Connection for the pool once one became invalid
- [ ] Make sure the rest of the code interacts with it in an ideal way
- [ ] Optimize the Manager

**Comment by Lol3rrr** [comment]
Currently the Client's respond-Task is causing problems because there is no way to stop it once the request is done but the connection is supposed to be kept open.
Looking into [AbortHandle](https://docs.rs/futures/0.3.12/futures/future/struct.Abortable.html) and [Abortable](https://docs.rs/futures/0.3.12/futures/future/struct.Abortable.html) may be worth while

**Comment by Lol3rrr** [comment]
Turns out that this wont work as expected because there are some unexpected difficulties, especially there is little control over how the underlying connections and protocols behave
